src/core/execute.c

```diff
@@ -4250,6 +4250,10 @@ static int exec_child(
                 }
 
                 r = cg_attach_everywhere(params->cgroup_supported, p, 0, NULL, NULL);
+                if (r == -EUCLEAN) {
+                        *exit_status = EXIT_CGROUP;
+                        return log_unit_error_errno(unit, r, "The threaded mode is detected, and could not attach process to cgroup %s.", p);
```

Comment:
maybe: "Unsupported use of threaded control group mode detected. Failed to attach process to cgroup '%s'."

Reply:
> maybe: "Unsupported use of threaded control group mode detected. Failed to attach process to cgroup '%s'."

Or maybe even `"Cgroup %s is threaded. Failed to attach process to cgroup: %m", p` ?

Reply:
The thing is that it's a sibling cgroup that is threaded not the one we are interested in. Hence the more vague wording I picked.

Reply:
Hmm, but doesn't the sibling group being threaded make this group threaded too?
`"Failed to attach process to cgroup %s because one of its siblings is threaded: %m", p` ?

Reply:
Maybe "parents" instead of "siblings"? Though, I am not familiar with the correct terms for cgroup...

Reply:
Siblings is the right term to use for cgroups that have the same parent cgroup. It's sprinkled throughout the cgroups documentation.

Reply:
Thanks. But from https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#threads, IIUC cgroup becomes "invald" state when its parent is in threaded mode. So maybe `"Failed to attach process to cgroup %s because the cgroup or one of its siblings or parent is in threaded mode: %m"`?

Reply:
sounds ok to me.